React/nested-event-handling.md

```diff
@@ -1,66 +1,89 @@
-- Continue to practice component creation and JSX syntax
-- Use `state` within our components
-- Act on events within our components
+
+By the end of this lesson, students should be able to...
+
+- Render child components from `state`
+- Understand when and how to lift state from a child component to a parent component
 - Pass event handler functions as callbacks within `props`
 
-### Overview
-In this discussion, we are going to go back to our students application and add some new features. First, we're going to refactor our code a bit to set ourselves up to use some of the new concepts we've learned about, like `state`. Then, we're going to add some new functionality so we can use our application to take attendance for our students.
+## Introduction
 
-## Refactor!
+Last time we worked on our student tracking application, we used a functional `StudentCollection` component to render a list of `Student`s. The student data came from an array created in the `StudentCollection` component.
 
-**This diagram will drive our overall approach for accomplishing this goal:**
-![nested components](images/nested-components.png)
-<!-- https://drive.google.com/open?id=1xq5jaCrI7FGp6PG1gr-bYE1ZTvPb5PxZ -->
+Our goal for this lesson is to make the application respond to events from the user. So, let's start with some brainstorming:
+- What kind of actions might the user take on individual students?
+- What kind of actions might the user take on the list of students?
 
-### Setup to manage list of data
-One student is great and all, but ideally, we'd really be able to manage our whole list of students that we might need. We'd also like to do this _outside_ of the `App` component, because what if our application also tracks Teachers?
+For this lesson our plan is to use our app to track attendance. Each student will have a button with the text "mark present", which when clicked, will show the student as present.
 
-We'll create a new component called `StudentCollection` that will live between the existing  `App` and `Student` components. This new component will manage the full list of students and then render the data for each student.
+## Refactor to a Stateful Component
 
-1. Spend a few minutes now going back to the [creating components](creating-components.md) notes to see how to create the new component file with the basic class, imports and exports you need.
+### Lifting State Up
 
-1. Move the code that renders the list of `Student`s from `App` to `StudentCollection`. Note: You will also need to move and update the `import` statement.
+Currently there is no way for us to change the student data. If we want to be able to change whether a student is present, we'll need to keep track of that as state in our application.
 
-1. Import and render `StudentCollection` now from the `App` component instead. Note: You will need to add an import statement.
+**Question:** Where should our application's state live? What are our options?
```

Comment:
I think that our students probably won't be able to even hypothesize an answer for this as they know they don't know the implications of where state should live -- As a lecturer I'd probably make this a rhetorical question at the moment.